pkg/build/apis/build/validation/validation_test.go

```diff
@@ -820,11 +820,14 @@ func TestValidateSource(t *testing.T) {
 	dockerfile := "FROM something"
 	invalidProxyAddress := "some!@#$%^&*()url"
 	errorCases := []struct {
-		t        field.ErrorType
-		path     string
-		source   *buildapi.BuildSource
-		ok       bool
-		multiple bool
+		t               field.ErrorType
+		path            string
+		source          *buildapi.BuildSource
+		ok              bool
+		multiple        bool
+		customStrategy  bool
+		dockerStrategy  bool
+		jenkinsStrategy bool
```

Comment:
Are these strategies mutually exclusive? I wonder if it is worth creating a "enum-like" type for the build strategy.

Reply:
they are mutually exclusive .. as for type, first blush, the value will be dependent on whether you include it in the actual api object def and change the signature to the point that is is a migration concern

I admit I'm not sure if it would be a concern, at least without visualizing the actual change ...

Bottom line, factor in migration concerns, whether you can dismiss them or not, in the decision making process